dependencies.yaml

```diff
@@ -258,7 +258,7 @@ dependencies:
           - *cmake_ver
           - cython>=3.0.3
           - *ninja
-          - &numpy numpy>=1.21
+          - &numpy numpy>=1.23
```

Comment:
Am just now noticing that we have NumPy as a build requirement here. However it is only used at build time by wheels. Conda packages don't have a build dependency on NumPy. So am wondering is the build requirement on NumPy actually needed? Or can it be dropped?

Reply:
After further discussion with Vyas offline and a bit of exploration, it looks like NumPy is used for its `halffloat.h` headers: https://github.com/rapidsai/cudf/pull/15165#issuecomment-1968090495

So this is needed